filecoin-hashers/src/poseidon.rs

```diff
@@ -362,8 +362,10 @@ impl LightAlgorithm<PoseidonDomain> for PoseidonFunction {
             1 | 2 | 4 | 8 | 16 => shared_hash_frs(
                 &parts
                     .iter()
-                    .map(|x| {
-                        <Bls12 as ff::ScalarEngine>::Fr::from_repr(x.0).expect("from_repr failure")
+                    .enumerate()
+                    .map(|(i, x)| {
```

Comment:
Anyone have a preference on this `enumerate`?  It was helpful with debugging, but if there are perf reasons to ditch it, it's no longer needed.

Reply:
should be fine